src/js/actions/transform.js

```diff
@@ -524,6 +476,17 @@ define(function (require, exports) {
             sizePromise = layerActionsUtil.playLayerActions(document, resizeLayerActions, true, options);
         }
 
+        // Reset type layers to pick up their implicit font size changes
+        sizePromise = sizePromise
+            .bind(this)
+            .then(function () {
+                var typeLayers = layerSpec.filter(function (layer) {
+                    return layer.kind === layer.layerKinds.TEXT;
+                });
+
+                return this.transfer(layerActions.resetLayers, document, typeLayers);
```

Comment:
`turns out` you may want to come back here after #2096 has merged, and tack on a third `true` param here.  Otherwise, this also suffers from the undo+redo combo ... such that the implied font size is not correctly restored.

I know this seems like it is getting *more complicated* instead of less, but this is helping at least to codify these edge cases where it makes sense to amend the current history. That was not a concept built in to the history store originally, so hopefully it can tightened up in round 2.